runtime-decompiler/src/main/java/org/fife/ui/hex/event/HexSearchActionListener.java

```diff
@@ -0,0 +1,46 @@
+package org.fife.ui.hex.event;
+
+import org.fife.ui.hex.swing.HexSearch;
+import org.jrd.backend.core.OutputController;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class HexSearchActionListener implements ActionListener {
+
+
+    private final HexSearch hexSearchEngine;
+    private final JTextField hexSearch;
+    private final JComboBox<HexSearch.HexSearchOptions> hexSearchType;
+    private final Method method;
+
+    public enum Method {
+        NEXT,
+        PREV
+    }
```

Comment:
This is usually beter to solve by abstract class - doing all the work - and two extensions of it - next and prev, which adds the minimal change whcih divergs here. But it si ok to keep like this for this case.